### PR TITLE
Extract fromToMotion utility function to reduce BarSeries animation boilerplate.

### PR DESCRIPTION
**Pull request by alantreadway**

### CODE REVIEW

File: packages/ag-charts-community/src/motion/fromToMotion.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+ * @param diff optional diff from a DataModel to use to detect added/moved/removed cases
+ */
+export function fromToMotion<N extends Shape, T extends AnimationValue & Partial<N>, D>(
+    id: string,
+    animationManager: AnimationManager,
+    selections: Selection<N, D>[],
+    fromFn: (node: N, datum: D, state: NodeUpdateState) => T,
+    toFn: (node: N, datum: D, state: NodeUpdateState) => T,
+    extraOpts: Partial<AnimationOptions<T> & AdditionalAnimationOptions> = {},
+    idFn?: (node: N, datum: D) => string,
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/motion/fromToMotion.ts]: `idFn` is called `getDatumId` everywhere else, probably worth keeping that aligned

File: packages/ag-charts-community/src/motion/fromToMotion.ts [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+        for (const node of selection.nodes()) {
+            let status: NodeUpdateState = 'added';
+            if (idFn && diff) {
+                status = calculateStatus(node, node.datum, idFn, ids);
+            }
+
+            cleanup ||= status === 'removed';
+
+            const from = typeof fromFn === 'function' ? fromFn(node, node.datum, status) : fromFn;
+            const to = typeof toFn === 'function' ? toFn(node, node.datum, status) : toFn;
```
Comment (lsjroberts) [inline packages/ag-charts-community/src/motion/fromToMotion.ts]: Are these `typeof` checks needed? Looks like they are always functions from the function param types.

**Review by lsjroberts — APPROVED** [review]
LGTM

File: packages/ag-charts-community/src/motion/fromToMotion.ts (hunk 1 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/motion/fromToMotion.ts]: Good call, done!

File: packages/ag-charts-community/src/motion/fromToMotion.ts (hunk 2 above)
Comment (alantreadway) [reply packages/ag-charts-community/src/motion/fromToMotion.ts]: Fair - removed.

**Pull request merged**